amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/property/like/MandatoryParser.scala

```diff
@@ -6,21 +6,77 @@ import amf.core.internal.parser.domain.{Annotations, ScalarNode}
 import amf.aml.internal.metamodel.domain.PropertyLikeMappingModel
 import amf.aml.client.scala.model.domain.PropertyLikeMapping
 import amf.aml.internal.parse.dialects.DialectContext
-import org.yaml.model.YMap
+import amf.core.internal.annotations.VirtualElement
+import org.yaml.model.{YMap, YMapEntry}
+
+import scala.language.implicitConversions
 
 case class MandatoryParser(map: YMap, propertyLikeMapping: PropertyLikeMapping[_ <: PropertyLikeMappingModel])(
     implicit val ctx: DialectContext) {
   def parse(): Unit = {
-    map.key(
-        "mandatory",
-        entry => {
-          val required = ScalarNode(entry.value).boolean().toBool
-          val value    = if (required) 1 else 0
-          propertyLikeMapping.set(propertyLikeMapping.meta.MinCount,
-                                  AmfScalar(value, Annotations(entry.value)),
-                                  Annotations(entry))
-        }
-    )
+
```

Comment:
To be able to see the cases better you could:
- extract each case into its own method. In that way, I can easily see all the cases with a method which in its name, clearly describes the functionality
- I wouldn't make each `if` clause be able to "interact" with the other clauses. Each case should be self-contained to be easier to explore and understand.